extmod/vfs_fat_file.c

```diff
@@ -139,39 +139,59 @@ STATIC const mp_arg_t file_open_args[] = {
 STATIC mp_obj_t file_open(fs_user_mount_t *vfs, const mp_obj_type_t *type, mp_arg_val_t *args) {
     int mode = 0;
     const char *mode_s = mp_obj_str_get_str(args[1].u_obj);
-    // TODO make sure only one of r, w, x, a, and b, t are specified
+    uint32_t rwxa_count = 0;
+    uint32_t bt_count = 0;
+    uint32_t plus_count = 0;
+    bool bad_mode = false;
     while (*mode_s) {
         switch (*mode_s++) {
             case 'r':
                 mode |= FA_READ;
+                rwxa_count++;
                 break;
             case 'w':
                 mode |= FA_WRITE | FA_CREATE_ALWAYS;
+                rwxa_count++;
                 break;
             case 'x':
                 mode |= FA_WRITE | FA_CREATE_NEW;
+                rwxa_count++;
                 break;
             case 'a':
                 mode |= FA_WRITE | FA_OPEN_ALWAYS;
+                rwxa_count++;
                 break;
             case '+':
                 mode |= FA_READ | FA_WRITE;
+                plus_count++;
                 break;
             #if MICROPY_PY_IO_FILEIO
             case 'b':
+                bt_count++;
                 type = &mp_type_vfs_fat_fileio;
                 break;
             #endif
             case 't':
+                bt_count++;
                 type = &mp_type_vfs_fat_textio;
                 break;
+            default:
+                bad_mode = true;
+                mp_raise_ValueError(translate("Invalid mode"));
```

Comment:
```suggestion
```

Rely on the `if` below for the error. That way it's raised in one spot.

Reply:
My typo! That's what I mean to do. Fixed.